lib/dialects/abstract/index.ts

```diff
@@ -0,0 +1,143 @@
+type DialectSupports = {
+  'DEFAULT': boolean,
+  'DEFAULT VALUES': boolean,
+  'VALUES ()': boolean,
+  'LIMIT ON UPDATE': boolean,
+  'ON DUPLICATE KEY': boolean,
+  'ORDER NULLS': boolean,
+  'UNION': boolean,
+  'UNION ALL': boolean,
+  'RIGHT JOIN': boolean,
+
+  /* does the dialect support returning values for inserted/updated fields */
```

Comment:
Do we need the comments on both parts? I would only keep these in the implementations

Reply:
I kept them in the typing only so they're shared by all implementations